test/api.test.js

```diff
@@ -0,0 +1,111 @@
+import { test } from './helpers/context.js'
+
+import git from 'git-rev-sync'
+import pWaitFor from 'p-wait-for'
+import delay from 'delay'
+import { toString } from 'uint8arrays/to-string'
+import { Dealer } from '@web3-storage/filecoin-client'
+import { randomAggregate } from '@web3-storage/filecoin-api/test'
+
+import { getDealerClientConfig } from './helpers/dealer-client.js'
+import { getTableItem } from './helpers/table.js'
+import { getBucketItem } from './helpers/bucket.js'
+import {
+  getApiEndpoint,
+  getStage,
+  getDealStoreDynamoDb,
+  getOfferStoreBucketInfo
+} from './helpers/deployment.js'
+
+test.before(t => {
+  t.context = {
+    apiEndpoint: getApiEndpoint(),
+    dealStoreDynamo: getDealStoreDynamoDb(),
+    offerStoreBucket: getOfferStoreBucketInfo()
+  }
+})
+
+test('GET /version', async t => {
+  const stage = getStage()
+  const response = await fetch(`${t.context.apiEndpoint}/version`)
+  t.is(response.status, 200)
+
+  const body = await response.json()
+  t.is(body.env, stage)
+  t.is(body.commit, git.long('.'))
+})
+
+test('POST /', async t => {
+  const {
+    invocationConfig,
+    connection
+  } = await getDealerClientConfig(new URL(t.context.apiEndpoint))
+
+  // Create a random aggregate
+  const label = 'label'
+  const { pieces, aggregate } = await randomAggregate(100, 128)
+  const offer = pieces.map((p) => p.link)
+
+  const res = await Dealer.dealQueue(
+    invocationConfig,
+    aggregate.link.link(),
+    offer,
+    invocationConfig.with,
+    label,
+    { connection }
+  )
+
+  t.truthy(res.out.ok)
+  t.falsy(res.out.error)
+  t.truthy(aggregate.link.link().equals(res.out.ok?.aggregate))
+
+  // wait for deal-store entry to exist given it is propagated with a queue message
+  await delay(5_000)
```

Comment:
Do we need this? Just increase the timeout below?

Reply:
it will take quite some time. For these 5 seconds is really not worth to even trying IMHO.. But we can use the `p-retry` `minTimeout` for it.